### DISCUSSION TITLE
Confirmed - Works on the Kurzweil K2600 series

### DISCUSSION THREAD

**Discussion by manpaz** [erichelgeson/BlueSCSI-v1#155] [Show and tell]
Hello,

I'd like to confirm that the BlueSCSI works on the Kurzweil K2600 series, I tried it on the following units and it worked with no problem:

- K2600X
- K2661

Tested witha a BlueSCSI 1.1a DB25

Based on the instructions on the Kurzweil's documentation for SCSI drives I created 4 drives of 2GB each, on the LUN0 and assigned them to the SCSI IDs 2,3,4, and 5 to avoid conflicts with the ID 6 which is the default for the Kurzweil instruments, ID 1 which is usually the default for internal HDs, and ID 0 for communications with external computers.

The procedure used is as follows:
1. On Linux, execute the following commands to produce the needed volumes:
```
dd if=/dev/zero of=HD20_512.hda bs=4M count=512
dd if=/dev/zero of=HD30_512.hda bs=4M count=512
dd if=/dev/zero of=HD40_512.hda bs=4M count=512
dd if=/dev/zero of=HD50_512.hda bs=4M count=512
```
2. Once the volumes are done, format the SD card as ExFAT and then copy over the images into the SD card.
3. Umount the SD card
4. Plug-in the SD card to the BlueSCSI, hook up the BlueSCSI to the Kurzweil and then power up the Kurzweil.
5. Once the unit is on, press the "DISK" button, and go to the SCSI ID 2
6. Format the volume, do not partition it, unless you created the volumes larger than 2GB
7. Repeat the steps 4 & 5 for SCSI IDs 3,4, and 5
8. Turn off the Kurzweil, dettach the BlueSCSI from the Kurzweil, take the SD card and put it back on the computer
9. Mount the SD Card
10. Mount the first volume  (Assuming that you labeled the SD Card `BlueSCSI`)
```
sudo mount /run/media/$(whoami)/BlueSCSI/HD20_512.hda /mnt/
```
11. Copy all your files into the `/mnt` directory
```
cp -r /path/to/my/samples /mnt/
```
13. Umount the volume
```
sudo umount /mnt
```
14. Umount the SD card
15. Plug-in the SD card to the BlueSCSI, hook up the BlueSCSI to the Kurzweil and then power up the Kurzweil
16. Turn the Kurzweil on again, press "DISK" button, and go to the SCSI ID 2
17. Press the soft button "Load" and you should be able to see your files and navigate through the volume on the SCSI 2
18. Select your samples & programs and load them
19. Play the samples at the selected location and there should be no corruption, ie, no crackling sounds, unless the sampling RAM is defective

Hope that helps others looking for a solution for their Kurzweil K2600.

Thank you!

**Comment by erichelgeson** [comment]
Thanks for taking the time to write down these instructions!

**Reply by manpaz** [reply]
Your very welcome, glad to help on a project that is also helping out others :)

**Reply by dmderev** [reply]
Manuel, 
Is there a similar way to do the partitioning and copy files on Windows (eg with some partition management software tool)?
Also, if I have the set of files on CD (they are *.krz), should I just copy these in any order into the partition? Or, if I have CD *.iso file, can I copy the entire file into a partition (I doubt if it works)?
When SD partition is created, does it work reliably for writing?

**Reply by dmderev** [reply]
Ok, I can answer my question: I am using PassMark OSFMount utility emulating the logical disk to mount the image. Then I am copying files. I also decided to create one blank file on SD, then format it in K2600 and copy it back to PC in several copies, Then I am mounting the local (not SD) image and copying necessary files. In some cases it makes sense to copy the mounted image to SD card without unmounting. And keeping formatted "blank" 2GB image file on PC simplifies the process of creating additional disk images without formatting. I used 6 disks so far: 00, 01,02, 03, 04 and 05.
Another observation: it looks that K2600 does not like nested directories and long filenames. I can open directories on PC when disk image is mounted, but K2600 does not open some (?) and instead returns to root of the disk. Still exploring the reason.

**Reply by manpaz** [reply]
Hey @dmderev ,
I'm sorry about the delay. About the files, yeah, you might need to create an image first, mount it in your local system and then dump the *.krz files in there. Obviously you have to umount the image once you are done :)
For the CD *.iso files, you can just copy them to the SD and rename that image to any of the HD<N>0_512.hda  available names in the SD, then the K2600 should be able to read the image.

**Reply by dmderev** [reply]
Manuel, Thanks!
One more remaining confusion. I collected the files from several CDs into a single 2GB image. It has two directory nesting levels, total 157 files. The files are present and not corrupted in the image. But when I try to access them on K260 through BlueSCSI, I see the entire root directory list, but cannot access files in some directories. When attempting to open them, they either open and show files, or instead return back to top directory list. The directories are displayed in alphabetic order, and the ones which I can open are on the top of the list. The directories fail to open not in the order of being written.
I tried to format it in DOS format on PC or format them in the synthesizer, the result is the same. BTW, now I am using ImDisk virtual driver on Windows. I also tried to create partitions of 2GB in larger image file, but the result is the same.
It appears that if the total size of the files written is below 1GB, all directories are accessible.
Did you experience something like? Could it be the problem of BlueSCSI (accessing the upper half of disk image file?). The user manual of K2600 says that it should support 2GB disks.

**Reply by dmderev** [reply]
Confirming that with <1GB on disk image, all files are visible for K2600. I do not have any other SCSI device to verify that upper sectors are accessible in 2GB images.

**Reply by manpaz** [reply]
Unfortunately, I don't have any large image as well, all I can think is to check with another SD.

**Reply by dmderev** [reply]
It is the same with any card and any format (formatting in synth or on PC via virtual drive). I also think that it should not be a hardware problem because all transactions go through the same 8-bit bus. It might be either a return of error when accessing above 1GB (a BlueSCSI firmware problem). It reveals as when attempting to open a directory which was written after 1GB was already copied to disk, the synth blinks and then displays the list of root directory. When read on PC, the virtual disk still shows all files correctly. So far I created virtual disks of 1GB in size, and there are no problems accessing them.

**Reply by bijlevel** [reply]
Thanks for these clear instructions! But before you can even start you have to realize that, at least for my K2500R, your (new) 50-pin scsi cable needs to be twisted before inserting it into the internal harddisk connector of the Kurzweil. To make this possible you have to remove the notch on the connector, otherwise it won't fit when twisted. I have added a picture of my K2500R with an old fashion scsi drive and you can clearly see the twisted cable. Odd, but true! ![IMG_0805](https://github.com/user-attachments/assets/535651f4-b9a8-4933-a4bb-0f4d0c6ee100)

**Comment by SynthfulDwarphus** [comment]
Thank you!
(I think you can create 4 drive partitions of 2GB each per SCSI ID on K2000, possibly bigger partitions on K25/K26.)

**Reply by manpaz** [reply]
Right, each drive could go up to 8GB so the Kurzweil can split each drive on 4 partitions of 2 GBs each. Then if you don't have anything else on the SCSI bus, you can use all IDs from 0 to 5 and 7, for a total of (6 drives) x (8 GB) = 48GB.
In my case I only had a 16GB SD card and no more than 4GB in samples, but still, that is more than enough to explore the rest of my life !!!

**Reply by SynthfulDwarphus** [reply]
Haha, yes, plenty of space! (Especially with such a synth that can do so much with so little!)
Thanks for your work on this.

Oh, is it possible to run CD-ROM ISOs on the K2 with BlueSCSI?

**Reply by manpaz** [reply]
I haven't tried yet, but you can try, let say dumping a CDRom on the SCSI ID 1, assuming the cd you want is mounted on the CD/DVD tray, with the following command:
```
dd if=/dev/cdrom of=HD10_512.hda bs=4M
```

**Reply by dmderev** [reply]
K2600 cannot use SCSI ID 7 as it is reserved to Mac computer. On the "partitions", it can create and format up to 4 partitions on a larger image file, but unfortunately, only Partition1 may be mounted on a PC.

**Reply by bijlevel** [reply]
The instructions provided for Linux to mount a .hda volume do not apply to a Mac, at least I get an error message:"invalid file system". What would be the right command(s) when using Terminal on a Mac? TIA!

**Reply by erichelgeson** [reply]
I do not know what file system the Kurzweil uses but if you just rename it to `.img` (bluescsi doesnt care about file extensions) and double click it on a modern mac, and it knows the file system, it will mount it.

**Reply by bijlevel** [reply]
@erichelgeson thanks! Changing the .hda extension to .img the Mac did mount that image and I could write files to that volume. Then I changed again the extension to .hda and the Kurzweil K2500R did show those files when de scsi drive was selected! This means, at least for a Mac, the steps above 9 written by @manpaz are not needed. Just changing the extension to .img, writing files to it, and changing it back to .hda is sufficient.

**Reply by erichelgeson** [reply]
>  and changing it back to .hda is sufficient.

There is no need to do this last step, file extensions are not used, just leave it .img :)

**Reply by bijlevel** [reply]
You're right, it works leaving it .img. Makes it even more easy! Thanks.

**Comment by SynthfulDwarphus** [comment]
Gotcha, and I agree that 4GB, and certainly 8GB, is a huge amount of storage for samples.

Yes, the other SCSI IDs might be useful if you're using a single SCSI2SD or something similar as a drive for multiple samplers simultaneously. 

On Aug 10, 2022, 12:25 AM, at 12:25 AM, Manuel F Martinez ***@***.***> wrote:
>Right, each drive could got up the 8GB so the Kurzweil can split each
>drive on 4 partitions of 2 GBs each. Then if you don't have anything
>else on the SCSI bus, you can use all IDs from 0 to 5 and 7, for a
>total of (6 drives) x (8 GB) = 48GB.
>In my case I only had a 16GB SD card and no more than 4GB in samples,
>but still, that is more than enough to explore the rest of my life !!!
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/erichelgeson/BlueSCSI/discussions/155#discussioncomment-3359552
>You are receiving this because you commented.
>
>Message ID:
***@***.***>
